R/loc_median_fit.R

```diff
@@ -27,39 +29,64 @@
 #'
 #'   plot(x, y)
 #'   fit <- loc_median_fit(x, y, fraction = 0.1)
+#'   fit2 <- loc_median_fit(x, y, fraction = 0.1, sample_fraction = 0.75)
 #'   points(x, fit, col = "red")
+#'   points(x, fit, col = "blue")
 #'
 #'
 #' @export
 loc_median_fit <- function(x, y, fraction = 0.1, npoints = max(1, round(length(x) * fraction)),
-                           weighted = TRUE, ignore_zeros = FALSE){
+                           weighted = TRUE, ignore_zeros = FALSE, sample_fraction = 1, seed=42){
+  if (length(x) != length(y)) {
+    stop("x and y must be of the same length.")
+  }
```

Comment:
This is certainly better than the original ` stopifnot(length(x) == length(y))`. However, it makes it also difficult to read because I cannot use the automatic diff view to see what are the relevant changes. 
It might be better to split the PR into two: the first would be to add the `sample_fraction` and the other to make these small convenience improvements.

Reply:
Yeah totally. I got a little carried away... As you pointed out below, I've refactored it to override `x` and `y`. This should hopefully make things much easier to read!